mobile/android/WirelessDebug/wirelessdebugger/src/main/java/com/google/wireless/debugger/WebSocketMessenger.java

```diff
@@ -170,15 +190,51 @@ public void sendSystemMetrics(int memUsed, int memTotal, double cpuUsage, double
             Log.e(TAG, e.toString());
         }
 
-        // TODO(Reece): Handle WebsocketNotConnectedException
-        send(payload.toString());
+        sendAndCatch(payload.toString());
     }
 
     /**
-     * Returns weather or not the connection is mRunning.
+     * Returns weather or not the connection is mIsRunning.
      * @return True if there is a connection, false otherwise.
      */
     public boolean isRunning() {
-        return mRunning;
+        return mIsRunning;
+    }
+
+    /**
+     * Sends a message over the websocket.  If the send fails or throws an exception, the message
+     * will be put back into the queue to try again later.  This function allows 10 retries of
+     * seding until it stops.
+     * @param message Message to be send.
+     * @param logsCopy Array List containing the original messages.  Needed to re-queue messages
+     *                 in the event of a failed send.
+     */
+    private void sendWithRetries(String message, ArrayList<String> logsCopy) {
+        try {
+            send(message);
```

Comment:
I don't think there is a way to tell if it succeeded without sending a message back from the server.  I can add a check to see if the connection is open and reset the counter then. It will have to check before sending though.